File: class_demo.cpp
#include <ctime>
#include <string>
using std::string;
int current_year()
{
    time_t now = time(NULL);
    tm *formatted = gmtime(&now);
    return formatted->tm_year + 1900;
}
class Person
{
public:
    Person();
    Person(const string first_name, const int year_born);
    int DOB;
private:
    string _first_name;
};
Person::Person()
{
}

Person::Person(const std::string first_name, const int year_born)
{
    DOB = year_born;
    _first_name = first_name;
}




int main(int argc, char *argv[])
{
    Person person1("Alice", 1980);
    Person person2;
    person2.DOB = 1986;
    person2._first_name = "Bob";

    return 0;
}

File: class_demo.py
from datetime import date
def current_year():
    return date.today().year
class Person:
    def __init__(self, first_name = None, DOB = None):
        self.first_name = first_name
        self.DOB = DOB

    def age(self):
        if self.DOB is not None:
            return current_year() - self.DOB
        return None

    def set_DOB(self, DOB):
        self.DOB = DOB

def main():
    person1 = Person('Alice', 1980)
    person2 = Person()
    person2.first_name = 'Bob'
    person2.set_DOB(1986)
    for person in (person1, person2):
        print 'Name: %s, Age: %d' % (person.first_name, person.age())

if __name__ == '__main__':
    main()

